libsolidity/ast/Types.h

```diff
@@ -881,6 +881,7 @@ class FunctionType: public Type
 	std::vector<std::string> const returnParameterTypeNames(bool _addDataLocation) const;
 	/// @returns the "self" parameter type for a bound function
 	TypePointer selfType() const;
+	virtual bool hasSelfType() const { return bound() && m_parameterTypes.size() > 0; }
```

Comment:
I think it should not be possible to construct a function type that is bound but does not have a self type.